ignite/metrics/psnr.py

```diff
@@ -0,0 +1,116 @@
+
+    .. math::
+        \text{PSNR}(i, j) = 10 * \log_{10}\left(\frac{ MAX_{i}^2 }{ \text{ MSE } }\right)
+
+    where :math:`\text{MSE}` is `mean squared error <https://en.wikipedia.org/wiki/Mean_squared_error>`_.
+
+    Note:
+        `y_pred` and `y` must have same dtype and same shape.
+        If `y_pred` and `y` have almost identical values, the result will be infinity.
+
+    Args:
+        data_range (int or float): The data range of the target image (distance between minimum
+            and maximum possible values). If not provided, it will be estimated from the input
+            data type. Typically, ``1.0`` for float tensor or ``255`` for unsigned 8-bit tensor.
+        output_transform (callable, optional): A callable that is used to transform the Engine’s
+            process_function’s output into the form expected by the metric.
+        device (str or torch.device): specifies which device updates are accumulated on.
+            Setting the metric’s device to be the same as your update arguments ensures
+            the update method is non-blocking. By default, CPU.
+
+    Example:
+
+    To use with ``Engine`` and ``process_function``, simply attach the metric instance to the engine.
+    The output of the engine's ``process_function`` needs to be in format of
+    ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y, ...}``.
+
+    .. code-block:: python
+
+        def process_function(engine, batch):
+            # ...
+            return y_pred, y
+        engine = Engine(process_function)
+        psnr = PSNR()
+        psnr.attach(engine, "psnr")
+        # ...
+        state = engine.run(data)
+        print(f"PSNR: {state.metrics['psnr']}")
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        data_range: Optional[Union[int, float]] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ):
+        super().__init__(output_transform=output_transform, device=device)
+        self.data_range = self.__data_range = data_range
+
+    def _check_shape_dtype(self, output: Sequence[torch.Tensor]) -> None:
+        y_pred, y = output
+        if y_pred.dtype != y.dtype:
+            raise TypeError(
+                f"Expected y_pred and y to have the same data type. Got y_pred: {y_pred.dtype} and y: {y.dtype}."
+            )
+
+        if y_pred.shape != y.shape:
+            raise ValueError(
+                f"Expected y_pred and y to have the same shape. Got y_pred: {y_pred.shape} and y: {y.shape}."
+            )
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self._sum_of_squared_error = 0  # type: Union[int, torch.Tensor]
+        self._num_elements = 0
+
+    @reinit__is_reduced
+    def update(self, output: Sequence[torch.Tensor]) -> None:
+        self._check_shape_dtype(output)
+        y_pred, y = output[0].detach(), output[1].detach()
+
+        if self.__data_range is None:
```

Comment:
Should we obligatory reintroduce `__data_range` , previous trick wont work anymore ?

Reply:
Since `self.data_range` is needed in `compute()`, I can't think of no other way
Let me know if there is.